### PR TITLE
Netex IDF: add trip properties

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

**Comment by woshilapin** [comment]
@datanel This PR needs a rebase.

File: src/netex_idf/stops.rs [hunk 1]
```diff
@@ -137,49 +129,25 @@ fn stop_point_parent_id(
...
-    })
-}
-
-fn get_or_create_equipment<'a>(
+pub fn get_or_create_equipment<'a>(
     quay: &Element,
-    equipments: &'a mut HashMap<StopAccessibility, Equipment>,
+    equipments: &'a mut HashMap<Accessibility, Equipment>,
     id_incr: &mut u8,
 ) -> Option<&'a mut Equipment> {
```
Comment (woshilapin) [inline src/netex_idf/stops.rs]: I don't think you need to return a mutable reference here.
```suggestion
) -> Option<&'a Equipment> {
```

File: src/netex_idf/offers.rs [hunk 2]
```diff
@@ -1089,6 +1091,7 @@ mod tests {
                 "Trip Short Name",
                 vehicle_journey.short_name.as_ref().unwrap()
             );
+            assert_eq!(Some("tp_id".to_string()), vehicle_journey.trip_property_id);
```
Comment (woshilapin) [inline src/netex_idf/offers.rs]:
```suggestion
            assert_eq!("tp_id", vehicle_journey.trip_property_id.as_ref().unwrap());
```

File: src/netex_idf/mod.rs [hunk 3]
```diff
@@ -20,6 +20,7 @@ mod lines;
 mod modes;
 mod offers;
 mod read;
+mod share;
```
Comment (woshilapin) [inline src/netex_idf/mod.rs]: How about calling it `accessibility` instead of `share`?

File: src/netex_idf/stops.rs [hunk 4]
```diff
@@ -137,49 +129,25 @@ fn stop_point_parent_id(
...
-    let accessibility = accessibility(quay)?;
+    let accessibility_node = quay.only_child("AccessibilityAssessment")?;
+    let accessibility = accessibility(accessibility_node)?;
 
-    let equipment = equipments.entry(accessibility).or_insert_with(|| {
-        let StopAccessibility {
-            wheelchair_boarding,
+    let equipment = equipments.entry(accessibility.clone()).or_insert_with(|| {
+        let Accessibility {
+            wheelchair,
```
Comment (woshilapin) [inline src/netex_idf/stops.rs]: You can also destructure with a different name for the variable.
```rust
let Accessibility {
            wheelchair: wheelchair_boarding,
            visual_announcement,
            audible_announcement,
        } = accessibility;
```
This way, you do not need to do the swap of names below on line 150. I know it's highly subjective but I do prefer to name correctly the variable early.

File: src/netex_idf/lines.rs [hunk 5]
```diff
@@ -63,12 +68,43 @@ fn line_color(line: &Element, child_name: &str) -> Option<Rgb> {
...
+) -> Option<&'a mut TripProperty> {
+    let accessibility_node = line.only_child("AccessibilityAssessment")?;
+    let id: String = accessibility_node.attribute("id")?;
+    let accessibility = accessibility(accessibility_node)?;
+
+    let trip_property = trip_propertiess
+        .entry(accessibility.clone())
+        .or_insert_with(|| {
+            let Accessibility {
+                wheelchair,
```
Comment (woshilapin) [inline src/netex_idf/lines.rs]: You can destructure and rename the variable.
```suggestion
                wheelchair: wheelchair_accessible,
```

File: src/netex_idf/stops.rs [hunk 6]
```diff
@@ -137,49 +129,25 @@ fn stop_point_parent_id(
...
             visual_announcement,
             audible_announcement,
+            ..
```
[the remark this answers is not in the document]
Comment (datanel) [reply src/netex_idf/stops.rs]: it's because I had 4 attributes in a previous version

[the remark this answers is not in the document]
Comment (woshilapin) [reply src/netex_idf/stops.rs]: I guessed :smile:

File: src/netex_idf/mod.rs (hunk 3 above)
Comment (datanel) [reply src/netex_idf/mod.rs]: i you want. I was planning to add more things

File: src/netex_idf/offers.rs (hunk 2 above)
Comment (datanel) [reply src/netex_idf/offers.rs]: why do you think it's better?

Comment (woshilapin) [reply src/netex_idf/offers.rs]: It's easier to read the expected part which to me is the important part.

Comment (datanel) [reply src/netex_idf/offers.rs]: I don't agree but as it's the same thing I will modify it

**Pull request merged**